devices/IWearRemapper/src/IWearRemapper.cpp

```diff
@@ -135,15 +136,38 @@ bool IWearRemapper::open(yarp::os::Searchable& config)
     }
 
     // RPC ports
-    if (!(config.check("wearableRPCPorts") && config.find("wearableRPCPorts").isList())) {
-        yError() << logPrefix << "wearableRPCPorts option does not exist or it is not a list";
-        return false;
+    if (!(config.check("useRPC"))) {
+        yWarning() << logPrefix << "OPTIONAL parameter useRPC NOT found";
     }
-    yarp::os::Bottle* inputRPCPortsNamesList = config.find("wearableRPCPorts").asList();
-    for (unsigned i = 0; i < inputRPCPortsNamesList->size(); ++i) {
-        if (!inputRPCPortsNamesList->get(i).isString()) {
-            yError() << logPrefix << "ith entry of wearableRPCPorts list is not a string";
-            return false;
+    else {
+        if (!config.find("useRPC").isBool()) {
+            yError() << logPrefix << "useRPC option is not a bool";
+        }
+        pImpl->useRPC = config.find("useRPC").asBool();
+        yWarning() << "useRPC set to" << pImpl->useRPC;
+    }
+
+    yarp::os::Bottle* inputRPCPortsNamesList = nullptr;
+    if (pImpl->useRPC)
+    {
+        if (!config.check("wearableRPCPorts"))
+        {
+            yInfo() << logPrefix << "wearableRPCPorts parameter does not exist. RPC port will not be used";
```

Comment:
> Maybe here I would `return false`. If `useRPC = true` the user wants to use RPC ports, and all the wrong options should make the device fail.

My idea was that if the parameter `useRPC` is not used, it is still possible to not using the `RPC` by omitting the `wearableRPCPorts` list. I don't know if this option can be helpful, or it makes things more confusing.

Reply:
I think the more intuitive use case is:

1. I want to use RPC ports. I have to use `useRPC` and specify the port names.
1. I do not want to use RPC ports. I can omit to add `useRPC` in the conf file, and as a consequence I can omit all other related options.

This approach will also allow to enable / disable RPC while keeping everything in the configuration.

Reply:
> 2\. I do not want to use RPC ports. I can omit to add `useRPC` in the conf file
 
@diegoferigo Isn't it better to have the option used in the config file and set it to `false`. This way a user will be aware of the option and its functionality directly from the config file instead of looking for it in the documentation

Reply:
Here we are talking about default values. Creating a xml configuration from a module should not be based on other configurations because that might refer to previous versions but it should always refer to the documentation, that must be updated.

Usually I handle it by printing at the end of the `configure` process a recap of how the module have been configured, including the default options. In this way the user has a real time feedback.

Reply:
> Usually I handle it by printing at the end of the `configure` process a recap of how the module have been configured, including the default options. In this way the user has a real time feedback.

Yes, the recap is being printed at the end of the configuration. 

I think that just have to be decided is how to handle the case in which `useRPC` is not explicit. Since, as discussed in the [comment below](https://github.com/robotology-playground/wearables/pull/24#discussion_r255020140), it may be completely removed in the future, I would say that we can omit `useRPC` in case we don't want to use it.

Reply:
Yes, thumbs up for `useRPC=false` by default

Reply:
> Yes, thumbs up for `useRPC=false` by default

Done https://github.com/robotology-playground/wearables/pull/24/commits/c71d6489e8bb3259bd3b0139ec3255c7691c5464, and tested.